c10/util/Half-inl.h

```diff
@@ -8,23 +8,41 @@
 
 #ifdef __CUDACC__
 #include <cuda_fp16.h>
+#define __NOT_CPU__
 #endif
 
 #ifdef __HIPCC__
 #include <hip/hip_fp16.h>
+#define __NOT_CPU__
 #endif
 
 #if defined(CL_SYCL_LANGUAGE_VERSION)
 #include <CL/sycl.hpp> // for SYCL 1.2.1
+#define __NOT_CPU__
 #elif defined(SYCL_LANGUAGE_VERSION)
 #include <sycl/sycl.hpp> // for SYCL 2020
+#define __NOT_CPU__
 #endif
 
 #if (defined(CPU_CAPABILITY_AVX2) || defined(CPU_CAPABILITY_AVX512)) && \
     !defined(__APPLE__)
 #include <ATen/cpu/vec/vec_half.h>
 #endif
 
+// including this stanza seemed like good hygiene, but it ends up messing
+// with the compiler's ability to figure out what the execution mode is below
+#if 0
+#ifndef __NOT_CPU__
+#if defined(__GNUC__) || defined(__clang__)
+#if defined(__aarch64__)
+#if (!defined(__ARM_32BIT_STATE)) || (__ARM_32BIT_STATE != 1)
+#define NATIVE_FP16 1
+#endif // __ARM_32BIT_STATE
+#endif // __aarch64__
+#endif // GNUC or clang
+#endif // __NOT_CPU__
+#endif
```

Comment:
What's the problem here?

Reply:
If Half-inl is included directly (not such inclusion currently in the code base), then the symbols defined in Half.h are not available.